### DISCUSSION TITLE
Can't Discover from Scoop

### DISCUSSION THREAD

**Discussion by UtopieStudio** [Devolutions/UniGetUI#1095] [General]
WUI shows my installed apps from Scoop, but it won't list anything from Scoop in the Discover section. Anyone know why?

Thanks.

**Comment by marticliment** [comment]
Can you please share your log?

Thanks!

**Reply by UtopieStudio** [reply]
Hi, sure thing.

`🟢 Using cached lang file
🟡 Using bundled lang file (forced=True)
It took 0.0019991397857666016 to load all language files
🟢 checkQueue Thread started!
🔵 Language file up-to-date
🔵 Starting main application...
Scanning for instances...
Found lock file(s), reactivating...
 * Serving Flask app 'WingetUI backend'
 * Debug mode: off
🔵 Starting update check
b'v1.4.10173\r\n'
None
updating winget
🟢 Downloaded latest metadata to local file
🔵 Loaded metadata from local file
b'2.0.0\r\n'
None
🔵 Version URL: https://versions.marticliment.com/versions/wingetui.ver
🟢 Updates not found
b"Current Scoop version:\nv0.3.1 - Released at 2022-11-15\n\n'extras' bucket:\n616d94362 telegram-downloader: Update to version 0.9.2\n\n'games' bucket:\nff6bed25 yuzu: Update to version 0-1464\n\n'main' bucket:\nd1fafb0da pandoc-crossref: Update to version 0.3.16.0a\n\n'sysinternals' bucket:\n3ad58eb sysinternals-suite@2023.03.30: Fix hash (Closes #12)\n\n"
None
Reactivation signal ignored: RaiseWindow_[1686411980.4153361]
{'wingetFound': True, 'scoopFound': True, 'chocoFound': True, 'sudoFound': True, 'wingetVersion': 'v1.4.10173\r', 'scoopVersion': 'v0.3.1 - Released at 2022-11-15', 'chocoVersion': '2.0.0\r', 'sudoVersion': 'gsudo v2.0.4 (Branch.tags-v2.0.4.Sha.506efa024af0cef6e4b0cfec42e0c8c5d0b1472c)\r'}
🔵 Reached main ui load milestone
Can't disconnect
🔵 Starting winget search
🔵 Starting scoop search
🔵 Starting choco search
🟢 Discover tab loaded
🟢 Found valid cache for winget!
🟢 Found valid cache for scoop!
🔵 Starting scoop file update...
🟢 Found valid cache for chocolatey!
Can't disconnect
Can't disconnect
🔵 Starting winget file update...
🟢 Starting winget search, winget on C:\Users\Dov\scoop\apps\WingetUI\current\winget-cli\winget.exe...
🟢 Starting scoop search...
🟢 Starting choco search, choco on choco.exe...
🟢 Upgrades tab loaded
Can't disconnect
🟢 Starting winget search, winget on C:\Users\Dov\scoop\apps\WingetUI\current\winget-cli\winget.exe...
🟢 Starting scoop search...
🟢 Starting choco search, choco on choco.exe...
🟢 Discover tab loaded
Can't disconnect
🟢 Settings tab loaded!
🟢 About tab loaded!
🟢 Chocolatey uninstallable packages search finished
🟢 Main application loaded...
Win32Mica Error: win32 version 19045 is not supported
50
🟢 Total packages: 5698
🟢 Searching for string ""
🟢 Scoop search finished
🟢 Searching for string ""
🟢 Winget search finished
🟡 Package Libretro.RetroArch version Unknown is ignored
🟡 Package Ubisoft.Connect is ignored
🟢 Searching for string ""
🟢 Winget uninstallable packages search finished
🟢 Searching for string ""
🟢 Searching for string ""
🟢 Total packages: 297
🟢 Chocolatey search finished
🟢 Searching for string ""
🟢 Scoop search finished
🟢 Searching for string ""
🟢 Searching for string ""
🟢 Total packages: 0
🟢 Searching for string "c"
🟢 Winget cache rebuilt
🟢 Total packages: 5698
🟢 Searching for string "cc"
🟢 Searching for string "ccl"
🟢 Searching for string "ccl"
🟢 Scoop cache rebuilt
🟢 Total packages: 5698`

**Comment by marticliment** [comment]
Hello,

There is a file called `ScoopCachedPackages` under YourUserDirectory\.wingetui\cacheddata. Can you please also share it?

**Reply by UtopieStudio** [reply]
There is no file with that name there, only one called scooppackages (and similar each for choco and winget). Here is that file contents though: [paste](https://pastebin.com/sqiP6Ebr)

**Reply by marticliment** [reply]
Can you try if with this beta release it does work better?

https://github.com/marticliment/WingetUI/releases/tag/2.0.0-beta3

**Reply by UtopieStudio** [reply]
Seems to be the same behaviour, but this time logs do make reference to cache not created being the issue:

```
🟡 Scoop cache file does not exist, creating cache forcefully and returning new package list
🔴 Could not load Scoop packages, returning an empty list!
```
I tried running in Admin mode, didn't make a difference.

**Reply by marticliment** [reply]
Can you please check the contents of the ScoopCachedPackages file?

Thanks!

**Reply by UtopieStudio** [reply]
So, CachePackages are both there for Winget and Choco this time, but still no file for Scoop.

**Reply by UtopieStudio** [reply]
Also, don't know if these error messages are of use:

```
🟢 Winget search for updates finished with 0 result(s) (msstore)
🟢 Searching for string "cpu-z"
🔴 Traceback (most recent call last):

🔴 Traceback (most recent call last):

🔴   File ".\PackageManagers\scoop.py", line 101, in cacheAvailablePackages

🔴   File ".\PackageManagers\scoop.py", line 101, in cacheAvailablePackages

🔴 IndexError: list index out of range

🔴 IndexError: list index out of range

🔴 Note this traceback was caught by reporter and has been added to the log (list index out of range)
🔵 Starting Scoop search for available packages
🟡 Scoop cache file does not exist, creating cache forcefully and returning new package list
🔴 Could not load Scoop packages, returning an empty list!
🟢 Total packages: 13966
```

**Reply by marticliment** [reply]
Yes, that's the cause of the issue!

Next WingetUI release (or beta release) will have this fixed!

**Comment by UtopieStudio** [comment]
Hi.

I updated to the latest version (2.0.1), enabled Scoop, re-launched and it still wasn't working, it also said in log log that Scoop wasn't the correct version and wasn't enabled. Checked for a populated ScoopCache file and it was indeed there . 

After repeating enable Scoop another 2-3 times and restarting the app, it suddenly started showing in Discovery searches.

Just thought I'd mention in case anyone else has the same behaviour.

Cheers.
